components/sidebar/add_channel_dropdown.tsx

```diff
@@ -47,6 +45,90 @@ const AddChannelDropdown = ({
     canCreateCustomGroups,
 }: Props) => {
     const intl = useIntl();
+    const dispatch = useDispatch();
+
+    const fallbackFocus = document.activeElement;
+
+    const returnFocus = () => {
+        document.dispatchEvent(new CustomEvent<A11yFocusEventDetail>(
+            A11yCustomEventTypes.FOCUS, {
+                detail: {
+                    target: fallbackFocus as HTMLElement,
+                    keyboardOnly: true,
+                },
+            },
+        ));
+    };
+
+    const openNewChannelModal = () => {
```

Comment:
These callbacks should be wrapped in `useCallback` to prevent the menu items from re-rendering when the `AddChannelDropdown` does